CatAnalyzer/plugins/TtbarDiLeptonAnalyzer.cc

```diff
@@ -270,8 +405,318 @@ bool TtbarDiLeptonAnalyzer::eventSelection(const edm::Event& iEvent, systematic
     b_genweight = (*genweightHandle);
     b_weight = b_genweight*b_puweight;
     
-    h_nevents->Fill(0.5,b_weight);
+    //h_nevents->Fill(0.5,b_weight);
+    h_nevents->Fill(0.5,b_genweight*b_puweight);
+  }  
+// #####################  Dstar begin  ########################
+  edm::Handle<cat::SecVertexCollection> d0s;       iEvent.getByToken(d0Token_,d0s);
+  edm::Handle<cat::SecVertexCollection> dstars;    iEvent.getByToken(dstarToken_,dstars);
+  edm::Handle<cat::SecVertexCollection> Jpsis;     iEvent.getByToken(JpsiToken_,Jpsis);
+
+  edm::Handle<edm::View<reco::GenParticle> > mcHandle;
+
+  if ( runOnMC ) {
+    iEvent.getByToken(mcSrc_, mcHandle);
+  }
+
+  vector<TLorentzVector> gen_d0s;
+  vector<TLorentzVector> gen_dstars;
+  vector<TLorentzVector> gen_Jpsis;
+
+  int nIDMother;
+
+  TLorentzVector vecSumMom;
+
+  if ( runOnMC ) {
+    for( const auto& aGenParticle : *mcHandle) {
+      //If genParticle is D0,
+      if ( std::abs(aGenParticle.pdgId()) == 421 ) {
+        gen_d0s.push_back( ToTLorentzVector(aGenParticle));
```

Comment:
Actually, Just 1 D0 will be selected for each jet on analysis step(on my thesis). However, To keep all D0 combination is required. Because D0 selection has a relation with D* meson. ( Distance between D0 and D*...)